browser/extensions/api/brave_rewards_api.cc

```diff
@@ -119,6 +119,43 @@ BraveRewardsTipTwitterUserFunction::Run() {
   return RespondNow(NoArguments());
 }
 
+BraveRewardsTipRedditUserFunction::BraveRewardsTipRedditUserFunction()
+    : weak_factory_(this) {
+}
+
+BraveRewardsTipRedditUserFunction::~BraveRewardsTipRedditUserFunction() {
+}
+
+ExtensionFunction::ResponseAction BraveRewardsTipRedditUserFunction::Run() {
+  std::unique_ptr<brave_rewards::TipRedditUser::Params> params(
+      brave_rewards::TipRedditUser::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  if (profile->IsOffTheRecord()) {
+    return RespondNow(
+        Error("Cannot tip Reddit user in a private context"));
+  }
+
+  RewardsService* rewards_service = RewardsServiceFactory::GetForProfile(
+      profile);
+
+  if (rewards_service) {
+    AddRef();
+    std::map<std::string, std::string> args;
+    args["user_name"] = params->reddit_meta_data.user_name;
+    args["post_text"] = params->reddit_meta_data.post_text;
+    args["post_rel_date"] = params->reddit_meta_data.post_rel_date;
+    rewards_service->SaveRedditPublisherInfo(
```

Comment:
I think we don't need a new function for this. We could have `rewards_service->SaveMediaInfo('reddit', args, bind)`. This way we don't need new function for every implementation.

Reply:
Will handle in refactor follow up: https://github.com/brave/brave-browser/issues/4907